packages/dds/tree/src/events/delegatingEventEmitter.ts

```diff
@@ -0,0 +1,113 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// eslint-disable-next-line import/no-nodejs-modules
+import { EventEmitter as NodeEventEmitter } from "events";
+import { EventEmitter, EventFilter } from "./events";
+
+/**
+ * Used by a {@link DelegatingEventEmitter} to store the underlying built-in event emitter
+ */
+export const wrappedEmitter = Symbol("The underlying event emitter");
+
+/**
+ * An {@link IEventEmitter} which delegates eventing to a built-in event emitter.
+ * Use the {@link wrappedEmitter} property to access the underlying emitter.
+ * Classes wishing to emit events may extend this class:
+ * @example
+ * ```typescript
+ * class MyClass extends DelegatingEventEmitter<MyEvents> {
+ *   // ...
+ * }
+ * ```
+ * Or, compose over it:
+ * @example
+ * ```typescript
+ * class MyClass extends IEventEmitter<MyEvents> {
+ *   private readonly events = DelegatingEventEmitter.create<MyEvents>();
+ *
+ *   public on<K extends (string | symbol) & keyof MyEvents>(
+ *     eventName: K,
+ *     listener: MyEvents[K],
+ *   ): () => void {
+ *     return events.on(eventName, listener);
+ *   }
+ * }
+ * ```
+ */
+export class DelegatingEventEmitter<
+    Events extends EventFilter<Events>,
+> extends EventEmitter<Events> {
```

Comment:
> If the pattern is successful, perhaps it can be adopted by other parts of Fluid

I think there are good ideas here, but it's questionable to innovate on a standardized interface in isolation.

Consider making Fluid adoption a prerequisite for SharedTree adoption?

Reply:
Might also be worth looking for preexisting API. Picking an existing standard saves us a lot of documentation, can improve interop and learning curve in addition to knowing the API works well in practice for others.

Fluid currently uses the API from NodeJS, which turns out to not work great with typescript as its target at JavaScript. I suspect there are a lot of good typescript alternatives out there already.

Reply:
Makes total sense.  FWIW - In the past we've prototyped API changes with SharedMap/Directory it's pretty simple, but has a lot of example usage.  Updating the examples was a pretty good measure of how a new API "feels" in practice.

Reply:
@CraigMacomber as I mentioned in the PR description, I did a cursory search and didn't find anything that was quite what I wanted, but that doesn't mean there isn't something out there.

@DLehenbauer awesome, great idea. If we do decide to pursue this further into the framework, that will be my starting point.